Git Files/Git Config.sublime-syntax

```diff
@@ -0,0 +1,185 @@
+        - match: \\
+          scope: invalid.unnecessary-escape.git.config
+        - include: illegal-newline
+
+  expect-section:
+    - match: (?=^\s*\[) # start of new section
+      pop: true
+    - match: \]
+      scope: punctuation.definition.brackets.end.git.config invalid.illegal.stray-bracket.git.config
+
+  # changed = red
+  # untracked = bold green
+  key-color-pair:
+    - match: ^\s*({{variable_name}})\s*(\=)
+      captures:
+        1: meta.mapping.key.git.config variable.other.readwrite.git.config
+        2: keyword.operator.assignment.git.config
+      push:
+        - meta_scope: meta.mapping.git.config
+        - meta_content_scope: meta.mapping.value.git.config
+        - include: expect-section
+        - include: expect-key
+        - include: color-value
+    - include: expect-section
+
+  # key = val
+  key-value-pair:
+    - match: ^\s*({{variable_name}})\s*(\=)
+      captures:
+        1: meta.mapping.key.git.config variable.other.readwrite.git.config
+        2: keyword.operator.assignment.git.config
+      push:
+        - meta_scope: meta.mapping.git.config
+        - meta_content_scope: meta.mapping.value.git.config
+        - include: expect-section
+        - include: expect-key
+        - include: other-value
+    - include: expect-section
+
+  expect-key:
+    - match: (?=^\s*{{variable_name}}\s*\=)  # next key-value pair
+      pop: true
+
+  color-value:
+    # example: bold, italic, underline
+    - match: \b(?:no-?)?(?:ul|strike|reverse|italic|dim|bold|blink)\b
+      scope: support.constant.color-attribute.git.config
+    # example: red, blue, green
+    - match: \b(?:yellow|white|red|normal|magenta|green|cyan|blue|black|auto)\b
+      scope: support.constant.color.git.config
+    # example: 0-255
+    - match: \b(?:{{zero_to_255}})\b
+      scope: constant.other.color.rgb-value.git.config
+
+  other-value:
+    - include: boolean
+    - include: string-quoted
+    - include: string-unquoted
+
+  boolean:
+    # 0 and 1 are omitted due to mismatches (e.g. "log -1", "HEAD~1"). They will
+    # be matched as unquoted strings instead.
+    - match: \b(?:yes|true|on|off|no|false)\b
+      scope: constant.language.git.config
+
+  string-quoted:
+    - match: \"
+      scope: punctuation.definition.string.begin.git.config
+      push:
+        - meta_include_prototype: false
+        - meta_scope: string.quoted.double.git.config
+        - match: \"
+          scope: punctuation.definition.string.end.git.config
+          set: expect-newline
+        - include: trailing-slash
+        - include: escape
+        - include: illegal-newline  # newlines only permitting after trailing \
+
+  string-unquoted:
+    - match: (?=\S)
+      push:
+        - meta_scope: string.unquoted.value.git.config
+        - include: trailing-slash
+        - include: escape
+        - match: $
+          pop: true
+
+  # The only valid escapes: '\b', '\n', '\t', '\"', '\\'.
+  escape:
+    - match: '%[A-Za-z]'
```

Comment:
Is this for the placeholders in [`--format=format:…`][1] or something more generic? If it's specifically for those placeholders, I can wrangle something to include the two-character ones and perhaps the more complicated ones as well.

[1]: https://git-scm.com/docs/pretty-formats

Reply:
As each config value can contain any valid shell script the format placeholder is not limited to `--format=...` but may include it. But you are right, not all git format patterns are handled by this pattern.